core/store/src/trie/trie_storage.rs

```diff
@@ -98,7 +98,11 @@ impl TrieCacheInner {
         is_view: bool,
     ) -> Self {
         assert!(cache_capacity > 0 && total_size_limit > 0);
-        let metrics_labels: [&str; 2] = [&shard_id.to_string(), if is_view { "1" } else { "0" }];
+        // `itoa` is much faster for printing shard_id to a string than trivial alternatives.
+        let mut buffer = itoa::Buffer::new();
+        let shard_id_str = buffer.format(shard_id);
+
+        let metrics_labels: [&str; 2] = [&shard_id_str, if is_view { "1" } else { "0" }];
```

Comment:
this is not a hot path, I don't think it makes sense to make the code more complicated here.